### ISSUE TITLE
Question / Issue: Writing to the BuildPack directory permissions issue

### ISSUE THREAD

**Issue by mattheworiordan** [heroku/heroku-buildpack-nodejs#193]
When using the Node JS build pack locally on our servers to build Node slugs, we hit a permission issue because our buildpack directory is immutable (read only).  We don't want to change that and we are not even sure why it should be changed, however in https://github.com/heroku/heroku-buildpack-nodejs/blob/master/lib/build.sh#L220-L224, it seems you write out the Node binary paths to the file export in the build pack directory.  We have had to workaround this issue with https://github.com/ably-forks/heroku-buildpack-nodejs/commit/c289da99406725cc0dc13d5e33ce5a92c5bbfb04 and everything works, but I'd like to understand why this is needed in the first place before I potentially create a PR to resolve this for others too.

Can someone advise why the buildpack is writing to the buildpack directory itself as opposed to the cache or build paths as it does everywhere else?

**Comment by hunterloftis** [comment]
Hi, this is a duplicate of https://github.com/heroku/heroku-buildpack-nodejs/issues/152.

tl;dr: the multi buildpack api has a new feature (exports) that enables multiple buildpacks to pass along environment variables. It writes to `exports` in the buildpack dir.

**Comment by mattheworiordan** [comment]
Thanks for the swift reply @hunterloftis, I completely understand the reasoning now.
